packages/cli/src/commands/pg/locks.ts

```diff
@@ -0,0 +1,56 @@
+import {Command, flags} from '@heroku-cli/command'
+import {Args, ux} from '@oclif/core'
+import {database} from '../../lib/pg/fetcher'
+import {exec} from '../../lib/pg/psql'
+import heredoc from 'tsheredoc'
+
+export default class Locks extends Command {
+  static topic = 'pg'
+  static description = 'display queries with active locks'
+  static flags = {
+    truncate: flags.boolean({char: 't', description: 'truncates queries to 40 characters'}),
+    app: flags.app({required: true}),
+    remote: flags.remote(),
+  }
+
+  static args = {
+    database: Args.string(),
+  }
+
+  public async run(): Promise<void> {
+    const {flags, args} = await this.parse(Locks)
+    const {app, truncate} = flags
+    const db = await database(this.heroku, app, args.database)
+    const query = heredoc`
+      SELECT
+        pg_stat_activity.pid,
+        pg_class.relname,
+        pg_locks.transactionid,
+        pg_locks.granted,
+        ${this.truncatedQueryString(truncate)} AS query_snippet,
+        age(now(), pg_stat_activity.query_start) AS "age"
+      FROM
+        pg_stat_activity,
+        pg_locks
+          LEFT OUTER JOIN pg_class
+          ON pg_locks.relation = pg_class.oid
+      WHERE
+        pg_stat_activity.query <> '<insufficient privilege>'
+        AND pg_locks.pid = pg_stat_activity.pid
+        AND pg_locks.mode = 'ExclusiveLock'
+        AND pg_stat_activity.pid <> pg_backend_pid() order by query_start;
+    `
+
+    const output = await exec(db, query)
+    ux.log(output)
+  }
+
+  protected truncatedQueryString(truncate: boolean) {
```

Comment:
nice improvement to this function. no idea why they were passing `prefix` as a parameter since it's literally just being used in this one place in this one command.

Reply:
I guessed it probably changed at some point in history or it was copied & pasted from somewhere.